pygmt/gridops.py

```diff
@@ -113,3 +113,115 @@ def grdcut(grid, **kwargs):
             result = None  # if user sets an outgrid, return None
 
         return result
+
+
+@fmt_docstring
+@use_alias(
+    G="outgrid",
+    F="filter",
+    D="distance",
+    I="increment",
+    N="nans",
+    R="region",
+    T="toggle",
+    V="verbose",
+    f="colinfo",
+)
+@kwargs_to_strings(R="sequence")
+def grdfilter(grid, **kwargs):
+    """
+    filter a grid file in the time domain using one of the selected convolution
+    or non-convolution isotropic or rectangular filters and compute distances
+    using Cartesian or Spherical geometries. The output grid file can optionally
+    be generated as a sub-region of the input (via *region*) and/or with new increment
+    (via *spacing*) or registration (via *toggle*).
+    In this way, one may have “extra space” in the input data so that the edges
+    will not be used and the output can be within one half-width of the input edges.
+    If the filter is low-pass, then the output may be less frequently sampled than the input.
+
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+        outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    filter : str
+        Name of filter type you which to apply, followed by the width
+        b: Box Car; c: Cosine Arch; g: Gaussian; o: Operator; m: Median;
+        p: Maximum Likelihood probability; h: histogram
+        Example: F='m600' for a median filter with width of 600
+    distance : str
+        Distance flag, that tells how grid (x,y) rrlated to the filter width as follows:
+        flag = p: grid (px,py) with width an odd number of pixels; Cartesian distances.
+
+        flag = 0: grid (x,y) same units as width, Cartesian distances.
+
+        flag = 1: grid (x,y) in degrees, width in kilometers, Cartesian distances.
+
+        flag = 2: grid (x,y) in degrees, width in km, dx scaled by cos(middle y), Cartesian distances.
+
+        The above options are fastest because they allow weight matrix to be computed only once.
+        The next three options are slower because they recompute weights for each latitude.
+
+        flag = 3: grid (x,y) in degrees, width in km, dx scaled by cosine(y), Cartesian distance calculation.
+
+        flag = 4: grid (x,y) in degrees, width in km, Spherical distance calculation.
+
+        flag = 5: grid (x,y) in Mercator -Jm1 img units, width in km, Spherical distance calculation.
```

Comment:
```suggestion
    filter : str
        ``xwidth[/width2][modifiers]``.
        Name of filter type you which to apply, followed by the width
        b: Box Car; c: Cosine Arch; g: Gaussian; o: Operator; m: Median;
        p: Maximum Likelihood probability; h: histogram
        Example: F='m600' for a median filter with width of 600
    distance : str
        Distance *flag* tells how grid (x,y) relates to filter width as
        follows:

        p: grid (px,py) with *width* an odd number of pixels; Cartesian
        distances.

        0: grid (x,y) same units as *width*, Cartesian distances.

        1: grid (x,y) in degrees, *width* in kilometers, Cartesian distances.

        2: grid (x,y) in degrees, *width* in km, dx scaled by cos(middle y),
        Cartesian distances.

        The above options are fastest because they allow weight matrix to be
        computed only once. The next three options are slower because they
        recompute weights for each latitude.

        3: grid (x,y) in degrees, *width* in km, dx scaled by cosine(y),
        Cartesian distance calculation.

        4: grid (x,y) in degrees, *width* in km, Spherical distance
        calculation.

        5: grid (x,y) in Mercator ``projection='m1'`` img units, *width* in km,
        Spherical distance calculation.
```
A few more style changes.